### ISSUE TITLE
HasSize matcher for Assertion in order to support the Collection and Iterable

### ISSUE THREAD

**Issue by dgroup** [llorllale/cactoos-matchers#74]
Now hamcrest has 2 different matchers in order to check the size of the collection (which is iterable) and iterable. Sometimes you can't use them easily without static methods from  `org.hamcrest.Matchers`:
 - org.hamcrest.Matchers#iterableWithSize(...)
 - org.hamcrest.Matchers#hasSize(...)

We are going to deprecate the static methods from `Matchers` thus we need to give an alternative to that behavior.

**Comment by llorllale** [comment]
@dgroup let's name it `HasSize`, and yes, it should directly accept a parameter of type `int`, not `Matcher<? super Integer>`

**Comment by llorllale** [comment]
@0crat in

**Comment by 0crat** [comment]
> @0crat in ([here](https://github.com/llorllale/cactoos-matchers/issues/74#issuecomment-452943089))

@llorllale Job [#74](https://github.com/llorllale/cactoos-matchers/issues/74) is now in scope, role is `DEV`

**Comment by Iprogrammerr** [comment]
@llorllale Could you assign me?

**Comment by llorllale** [comment]
@0crat assign @Iprogrammerr

**Comment by 0crat** [comment]
> @0crat assign @Iprogrammerr  ([here](https://github.com/llorllale/cactoos-matchers/issues/74#issuecomment-456620145))

@llorllale The job [#74](https://github.com/llorllale/cactoos-matchers/issues/74) assigned to @iprogrammerr[/z](https://www.0crat.com/u/iprogrammerr), here is [why](https://www.0crat.com/footprint/C63314D6Z/305715d0-fd9e-4823-8ddd-d0654125ffb1); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by Iprogrammerr** [comment]
@0crat waiting on outcome of #75

**Comment by 0crat** [comment]
> @0crat waiting on outcome of #75  ([here](https://github.com/llorllale/cactoos-matchers/issues/74#issuecomment-456977203))

@Iprogrammerr The impediment for [#74](https://github.com/llorllale/cactoos-matchers/issues/74) was registered successfully by @iprogrammerr[/z](https://www.0crat.com/u/iprogrammerr)

**Comment by 0crat** [comment]
The job [#74](https://github.com/llorllale/cactoos-matchers/issues/74) assigned to @victornoel[/z](https://www.0crat.com/u/victornoel), here is [why](https://www.0crat.com/footprint/C63314D6Z/f641c273-acdb-473a-8817-3b5ccd2ed212); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by victornoel** [comment]
@dgroup it seems @Iprogrammerr forgot to ask for closing the issue, I think it is solved and there are tests, can you close it now?

**Comment by victornoel** [comment]
@0crat wait for reporter

**Comment by 0crat** [comment]
> @0crat wait for reporter ([here](https://github.com/llorllale/cactoos-matchers/issues/74#issuecomment-480494847))

@victornoel The impediment for [#74](https://github.com/llorllale/cactoos-matchers/issues/74) was registered successfully by @victornoel[/z](https://www.0crat.com/u/victornoel)

**Comment by dgroup** [comment]
@victornoel thanks for reminding.

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @victornoel[/z](https://www.0crat.com/u/victornoel), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#74](https://github.com/llorllale/cactoos-matchers/issues/74) is now out of scope

**Comment by sereshqua** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +35 point(s) just awarded to @victornoel[/z](https://www.0crat.com/u/victornoel)

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)
